javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -519,6 +519,375 @@ object Character {
+    override final def hashCode(): scala.Int = super.hashCode
+    override final def toString(): String = name
+  }
+
+  final class UnicodeBlock private (name: String, private val start: Int, private val end: Int) extends Subset(name)
+
+  object UnicodeBlock {
+    // Initial size from script below
+    private[this] val allBlocks: ArrayList[UnicodeBlock] = new ArrayList[UnicodeBlock](221)
+    private[this] val blocksByNormalizedName = new HashMap[String, UnicodeBlock]()
+
+    private[this] def addNameAliases(properName: String, block: UnicodeBlock,
+        includeJVM: scala.Boolean = true): Unit = {
+      // Add normalized aliases
+      val lower = properName.toLowerCase
+      //   lowercase + spaces
+      blocksByNormalizedName.put(lower, block)
+      //   lowercase + no spaces
+      blocksByNormalizedName.put(lower.replace(" ", ""), block)
+      //   tokens as _
+      if (includeJVM) {
+        blocksByNormalizedName.put(
+            lower.replace(' ', '_')
+              .replace('-', '_'), block)
+      }
+    }
+
+    private[this] def addUnicodeBlock(properName: String, start: Int, end: Int): UnicodeBlock =  {
+      val jvmName = properName.toUpperCase
+        .replace(' ', '_')
+        .replace('-', '_')
+
+      val block = new UnicodeBlock(jvmName, start, end)
+      allBlocks.add(block)
+      addNameAliases(properName, block)
+      block
+    }
+
+    // Special JVM Constant, don't add to allBlocks
+    val SURROGATES_AREA = new UnicodeBlock("SURROGATES_AREA", 0x0, 0x0)
+    blocksByNormalizedName.put("surrogates_area", SURROGATES_AREA)
+
+/*
+    // Get the "proper name" for JVM block name
+    val blockNameMap: Map[String, String] = {
+      val blocksSourceURL = new java.net.URL("http://unicode.org/Public/UCD/latest/ucd/Blocks.txt")
+      val source = scala.io.Source.fromURL(blocksSourceURL, "UTF-8")
+      source
+        .getLines()
+        .filterNot {
+          _.startsWith("#")
+        }
+        .flatMap { line =>
+          line.split(';') match {
+            case Array(_, name) =>
+              val trimmed = name.trim
+              val jvmName = trimmed.replaceAll(raw"[\s\-]", "_").toUpperCase
+              Some(jvmName -> trimmed)
+            case _ => None
+          }
+        }.toMap ++ Seq(
+          "GREEK" -> "Greek",
+          "CYRILLIC_SUPPLEMENTARY" -> "Cyrillic Supplementary",
+          "COMBINING_MARKS_FOR_SYMBOLS" -> "Combining Marks For Symbols"
+        )
+    }
+
+    val blocksAndCharacters = (0 to Character.MAX_CODE_POINT)
+      .map(cp => Character.UnicodeBlock.of(cp) -> cp).filterNot(_._1 == null)
+
+    val orderedBlocks = blocksAndCharacters.map{ _._1 }.distinct
+
+    val blockLowAndHighCodePointsMap = {
+      blocksAndCharacters.groupBy{ _._1 }.mapValues { v =>
+        val codePoints = v.map{ _._2 }
+        (codePoints.min, codePoints.max)
+      }
+    }
+
+    // SURROGATES_AREA
+    val specialCounts = 1
```

Comment:
I thought `SURROGATES_AREA` is *not* part of `allBlocks, so it doesn't need to go into the count?